add NETC support on imx95 A55

JiafeiPan opened the pull request:
This PR is to enable NETC support on i.MX 95 A55.

This PR depends on hal PR: https://github.com/zephyrproject-rtos/hal_nxp/pull/571

JiafeiPan committed:
soc: imx95: a55: add netc power and clock init in soc.c

Power up NETCMIX and configure netc clock in soc_init().

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
dts: arm64: imx95_a55: add SCMI power device node

Added SCMI power dts node.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
dts: arm64: imx95_evk: add netc dts nodes

Added NETC dts nodes in imx95_evk A55 platform.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: imx95_evk: a55: add NETC support

Added NETC PSI0 support on the i.MX 95 EVK board for A55 platform.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
west.yml: depend on hal_nxp PR

Tmp patch to depend on hal_nxp PR 571.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan commented on drivers/ethernet/nxp_imx_netc/CMakeLists.txt:
```diff
@@ -4,6 +4,7 @@
 if(CONFIG_ETH_NXP_IMX_NETC)
   zephyr_library_sources(eth_nxp_imx_netc.c)
   zephyr_library_sources(eth_nxp_imx_netc_psi.c)
+  zephyr_library_sources_ifdef(CONFIG_ETH_NXP_IMX_NETC_MSI_GIC	eth_nxp_imx_netc_blk.c)
```
No, as you can find I moved these code from [board.c ](https://github.com/zephyrproject-rtos/zephyr/pull/92585/files/26c8fa848f0b62124240aef3ad3907fba504c9a9#diff-c2cfce380e220ee03785a1c0ed82daa91e6a055d4351c6921891adad57ff3661L13) to this block driver both for m-core and a-core.

JiafeiPan commented on drivers/ethernet/nxp_imx_netc/eth_nxp_imx_netc_blk.c:
```diff
@@ -0,0 +1,209 @@
+	DEVICE_DT_INST_DEFINE(inst, eth_nxp_imx_netc_blk_init, NULL,                               \
+			      &eth_nxp_imx_netc_blk_data_##inst,                                   \
+			      &eth_nxp_imx_netc_blk_config_##inst, POST_KERNEL,                    \
+			      CONFIG_MDIO_INIT_PRIORITY, NULL);
```
the init sequence should be (earlier to later)
eth_nxp_imx_netc_blk: CONFIG_MDIO_INIT_PRIORITY
mdio_nxp_imx_netc: CONFIG_MDIO_INIT_PRIORITY
eth_nxp_imx_netc: CONFIG_ETH_INIT_PRIORITY
eth_nxp_imx_netc_psi: CONFIG_ETH_INIT_PRIORITY

As block driver need to do some basic block initialization, so it need to be initialized firstly, then MDIO, PHY, NETC, PSI.
For these drivers which have the same initialization priority, the init sequence will defined according to dts nodes' definition, refer to [a55 netc dts](https://github.com/zephyrproject-rtos/zephyr/pull/92585/files/26c8fa848f0b62124240aef3ad3907fba504c9a9#diff-bd47b0b9a0fb1a86841d446c53c6c419737bfbd40b8675ae4566092854b443ceR486), MDIO is defined inside of netc blk, so MDIO will be initalized firstly although they have the same priority CONFIG_MDIO_INIT_PRIORITY, it is the same for netc and netc_psi.

yangbolu1991 replied:
It's clear. Thanks:)

JiafeiPan committed:
samples: net: zperf: add imx95 evk a55 suppport

Add i.MX 95 EVK A55 support for zperf sample.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
boards: imx95_evk: a55: enlarge networking stack size

The issue is caused by default stack size 1500 bytes of NET_RX_STACK_SIZE
is to small for ARM64 platform, then the stack of net_rx thread is
overflow and break the net_mgmt thread's stack, as a result net_mgmt thread
is crashed.

So this patch will enlarge all the stack size of net stack threads.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan commented:
v4: moved pci config memory region from netc dts node to each psi dts node as some platform such as imx943 can't use a continues region to cover all pci config region for all PSI's, it is also aligned with MDIO which we put config region in itself dts node. so that we could remove reg property from netc dts node.

JiafeiPan commented:
v5: rebased as GIC and NETC sub-PR have been merged.
